### ISSUE TITLE
Allow custom rules

### ISSUE THREAD

**Issue by chrismbarr** [twbs/bootlint#370]
The ability to add custom linting rules would be very amazing and useful.  It's very simple to write validation rule for this project, and I'd love the ability to extend this into custom rules for things that are specific to my project and not necessarily something that everyone using bootlint would benefit from.

The TSLint project allows for custom rules, and it's something that our organization has found extremely useful: https://github.com/palantir/tslint#custom-rules

**Comment by cvrebert** [comment]
On the one hand, Bootlint wasn't intended as a general-purpose HTML linter and there seem to be existing projects in that area, albeit with somewhat different approaches, e.g. https://github.com/curtisj44/HTML-Lint , https://github.com/htmllint/htmllint
On the other hand, it would be fairly easy to add such a feature (basically, make a linter class instead of a "singleton" module).

@twbs/bootlint Anyone have thoughts on this?

**Comment by chrismbarr** [comment]
That [htmllint project](https://github.com/htmllint/htmllint) looks promising & useful, thanks for pointing that out to me!  However it doesn't seem to allow for adding custom rules that exist in a project.

I totally understand that boot lint's purpose is not to validate HTML, but it's to validate the usage of the BootStrap framework.  But... this project makes writing rules so simple, mostly due to the face that jQuery selectors are to easy to write.  The way projects like TSLint handle this is by just pointing to a directory full of rules that follow a standard format.

This is 100% pseudo-code, but I could envision it working something like this:
```javascript
bootlint.lintHtml("<!DOCTYPE html><html>...", reporter, [], "path/to/custom-rules/");
```

Then that folder could contain JS files that could be loaded, each one having a function that could work just like the rules already built into the project.
```javascript
function rule($, reporter) {
  this.id = "C001";
  var blinks = $("blink");
  if(blinks.length > 0) {
    reporter("<blink> tags are not allowed!");
  }
}
```

It's just an idea, it seems like it would be useful, but also possibly unrelated to the goals of this project.  I'd love for something like this to exit either here, or somewhere else.
